contributed_definitions/NXreflections.nxdl.xml

```diff
@@ -0,0 +1,415 @@
+  <group type="NXentry" minOccurs="1">
+    
+    <field name="definition">
+      <doc> NeXus NXDL schema to which this file conforms </doc>
+      <enumeration>
+        <item value="NXreflections" />
+      </enumeration>
+    </field>
+
+    <field name="experiments" type="NXmx" minOccurs="1">
+      <doc> The experiments from which the reflection data derives </doc>
+    </field>
+
+    <field name="h" type="NX_INT" minOccurs="1">
+      <attribute name="description">
+        <doc> Describes the dataset </doc>
+      </attribute>
+      <doc> The h component of the miller index </doc>
+    </field>
+
+    <field name="k" type="NX_INT" minOccurs="1">
+      <attribute name="description">
+        <doc> Describes the dataset </doc>
+      </attribute>
+      <doc> The k component of the miller index </doc>
+    </field>
+
+    <field name="l" type="NX_INT" minOccurs="1">
+      <attribute name="description">
+        <doc> Describes the dataset </doc>
+      </attribute>
+      <doc> The l component of the miller index </doc>
+    </field>
+
+    <field name="id" type="NX_INT" minOccurs="1">
+      <attribute name="description">
+        <doc> Describes the dataset </doc>
+      </attribute>
+      <doc> 
+        The id of the experiment which resulted in the reflection. If the value
+        is greater than 0, the experiments must link to a multi-experiment NXmx
+        group 
+      </doc>
+    </field>
+
+    <field name="reflection_id" type="NX_INT" minOccurs="1">
+      <attribute name="description">
+        <doc> Describes the dataset </doc>
+      </attribute>
+      <doc>
+        The id of the reflection. Multiple partials from the same reflection
+        should all have the same id
+      </doc>
+    </field>
+    <field name="entering" type="NX_BOOL" minOccurs="1">
+      <attribute name="description">
+        <doc> Describes the dataset </doc>
+      </attribute>
+      <doc> Is the reflection entering or exiting the Ewald sphere </doc>
+    </field>
+    
+    <field name="det_module" type="NX_INT" minOccurs="1">
+      <attribute name="description">
+        <doc> Describes the dataset </doc>
+      </attribute>
+      <doc> The detector module on which the reflection was recorded </doc>
+    </field>
+
+    <field name="flags" type="NX_INT" minOccurs="1">
+      <attribute name="description">
+        <doc> Describes the dataset </doc>
+      </attribute>
+      <doc> Status flags describing the reflection </doc>
+    </field>
+
+    <field name="d" type="NX_FLOAT" minOccurs="1">
+      <attribute name="description">
+        <doc> Describes the dataset </doc>
+      </attribute>
+      <doc> The resolution of the reflection </doc>
+    </field>
+    
+    <field name="partiality" type="NX_FLOAT" minOccurs="1">
+      <attribute name="description">
+        <doc> Describes the dataset </doc>
+      </attribute>
+      <doc> The partiality of the reflection </doc>
+    </field>
+
```

Comment:
Don't specify the actual units in the NXDL file.  Specify the *type* of the units.  In this case, it is likely a sequence number of some sort?  In that case, why not ``NX_UNITLESS`` (or one from the above list)?

    lxml.etree.DocumentInvalid: Element '{http://definition.nexusformat.org/nxdl/3.1}field', attribute 'units': 
    [facet 'enumeration'] The value 'frames' is not an element of the set 
    {'NX_ANGLE', 'NX_ANY', 'NX_AREA', 'NX_CROSS_SECTION', 'NX_CHARGE', 
      'NX_CURRENT', 'NX_DIMENSIONLESS', 'NX_EMITTANCE', 'NX_ENERGY', 
      'NX_FLUX', 'NX_FREQUENCY', 'NX_LENGTH', 'NX_MASS', 'NX_MASS_DENSITY', 
      'NX_MOLECULAR_WEIGHT', 'NX_PER_AREA', 'NX_PER_LENGTH', 'NX_PERIOD', 
      'NX_POWER', 'NX_PRESSURE', 'NX_PULSES', 'NX_SCATTERING_LENGTH_DENSITY', 
      'NX_SOLID_ANGLE', 'NX_TEMPERATURE', 'NX_TIME', 'NX_TIME_OF_FLIGHT', 
      'NX_VOLTAGE', 'NX_VOLUME', 'NX_UNITLESS', 'NX_WAVELENGTH', 'NX_WAVENUMBER'}., 
    line 102